src/orca/services/nextflowtower/client.py

```diff
@@ -369,3 +371,47 @@ def list_workflows(
         json = self.get(path=path, params=params)
         items = self.unwrap(json, "workflows")
         return [models.Workflow.from_json(item["workflow"]) for item in items]
+
+    def get_workflow_tasks(
+        self,
+        workflow_id: str,
+        workspace_id: Optional[int] = None,
+    ) -> list[models.WorkflowTask]:
+        """Retrieve the details of a workflow run's tasks.
+
+        Args:
+            workflow_id: The ID number for a workflow run to
+            get tasks from.
+            workspace_id: The ID number of the workspace the workflow
+                exists within. Defaults to None.
+
+        Returns:
+            List of WorkflowTask objects.
+        """
+        path = f"/workflow/{workflow_id}/tasks"
+        params = self.generate_params(workspace_id)
+        json = self.get(path=path, params=params)
+        items = self.unwrap(json, "tasks")
+        return [models.WorkflowTask.from_json(item["task"]) for item in items]
+
+    def get_task_logs(
```

Comment:
One weakness of this feature is that you have to make 1 API call per task to get the logs. This could make it difficult to retrieve thousands of logs in a timely fashion without parallelization

Reply:
Would the other method be to just get the logs directly via AWS API?

Reply:
Yes, I did look into this alternative method:

- get tasks using `get_workflow_tasks`
- loop though tasks to get `work_dir` and use `s3fs` or another AWS-using method to download `command.log` from the bucket
- read contents of files into memory to process strings

But, I found this to be at least as inefficient if not worse. Mainly because of the need to actually download files which would require a complicated implementation since we would not want the potential for thousands of files to be downloaded to a local disk at one time.